packages/sources/layer2-sequencer-health/src/starkware.ts

```diff
@@ -0,0 +1,140 @@
+  const starkKeyPair = ec.genKeyPair(DEFAULT_PRIVATE_KEY)
+  const starkKeyPub = ec.getStarkKey(starkKeyPair)
+
+  ProviderInterface
+
+  const provider = new Provider({
+    baseUrl: config.starkwareConfig.sequencerUrl,
+    feederGatewayUrl: config.starkwareConfig.feederGatewayUrl,
+    gatewayUrl: config.starkwareConfig.gatewayUrl,
+  })
+
+  const account = new Account(provider, config.starkwareConfig.argentAccountAddr, starkKeyPair)
+
+  const receipt = await race<AddTransactionResponse>({
+    timeout: config.timeoutLimit,
+    promise: account.execute(
+      {
+        contractAddress: config.starkwareConfig.argentAccountAddr,
+        entrypoint: 'initialize',
+        calldata: [starkKeyPub, '0'],
+      },
+      undefined,
+      { maxFee: '0' },
+    ),
+    error: `Transaction receipt not received in ${config.timeoutLimit} milliseconds`,
+  })
+  Logger.info(
+    `Transaction receipt received with hash ${receipt.transaction_hash} for EVM network: ${Networks.Starkware}`,
+  )
+  await provider.waitForTransaction(receipt.transaction_hash)
+  const tx = await provider.getTransaction(receipt.transaction_hash)
+  return tx.status !== 'REJECTED'
+}
+
+/**
+ * The Starkware sequencer is made up of two internal components.  The first is the gateway which is used to query the Sequencer state
+ * and the batcher which adds transactions to the pending block and processes the pending block to add it into the L2 chain.  The
+ * Sequencer is considered unhealthy if either of the components are down.
+ *
+ *  - The gateway is considered to be down if request to fetch pending block times out or if we get a 404
+ *  - The batcher is considered to be down if we a new block has not been produced AND no new transactions have been added to the
+ *  pending block within a given time interval.
+ * status when fetching
+ * @param networks
+ * @returns true if Sequencer is healthy
+ */
+export const checkStarkwareSequencerPendingTransactions = (): ((
+  config: ExtendedConfig,
+) => Promise<boolean>) => {
+  let lastSeenPendingBlock: PendingBlockState = {
+    parentBlockHash: '',
+    txnCount: 0,
+    lastUpdated: 0,
+    lastStatus: true,
+  }
+  return async (config: ExtendedConfig): Promise<boolean> => {
+    const delta = config.delta
+    const currentTime = Date.now()
+    if (
+      lastSeenPendingBlock.lastUpdated > 0 &&
+      currentTime - lastSeenPendingBlock.lastUpdated < delta
+    ) {
+      Logger.debug(
+        `Skipping check for Starkware Sequencer health as it has been less than ${delta} seconds since last check`,
+      )
+      return lastSeenPendingBlock.lastStatus
+    }
+    let pendingBlockParams: StarkwareBlockResponse
+    try {
+      const { data } = await Requester.request<StarkwareBlockResponse>({
+        url: `${config.starkwareConfig.sequencerUrl}/${config.starkwareConfig.feederGatewayUrl}/get_block?blockNumber=pending`,
```

Comment:
so I'm initializing a provider instance somewhere else like this

```
const provider = new Provider({
    baseUrl: config.starkwareConfig.sequencerUrl,
    feederGatewayUrl: config.starkwareConfig.feederGatewayUrl,
    gatewayUrl: config.starkwareConfig.gatewayUrl,
  })
```

is this ok with you?  The `RPCProvider` class does not seem to be included in the latest `3.18.2` release

Reply:
Yeah, I've asked ShardLabs in the shared channel about the next release.

I'd prefer we migrate to the RPCProvider.

In the meantime, we can try pulling from git directly: https://docs.npmjs.com/cli/v6/configuring-npm/package-json#git-urls-as-dependencies

Reply:
Version [4.1.0](https://www.npmjs.com/package/starknet/v/4.1.0) is available on NPM.

Reply:
hmm I need to try this but if I'm not wrong we need to ping the Sequencer directly to get the list of pending transactions.  This is because they haven't implemented a gossip protocol to obtain the info from full nodes.  Can we use `SequencerProvider` instead for now?